main.tf

```diff
@@ -151,7 +151,8 @@ data "template_file" "user_data_client" {
 # ---------------------------------------------------------------------------------------------------------------------
 
 data "aws_vpc" "default" {
-  default = true
+  default = "${var.vpc_id == "" ? true : false}"
+  id = "${var.vpc_id == "" ? "" : var.vpc_id}"
```

Comment:
Couldn't you just set the `id` param to `var.vpc_id` without the if-statement? It seems like it would have no effect.

Also, does the `aws_vpc` data source work correctly if you explicitly set `id` to an empty string?

Reply:
@brikis98, you're absolutely right about the `id` param.  (facepalm)

I believe the data source does work correctly.  I ran terraform both explicitly specifying the `vpc_id` in the module as well as omitting the parameter.  In the former case the instances were associated with the specified vpc, and in the latter, the default vpc was used.